app/experimenter/experiments/serializers.py

```diff
@@ -252,9 +252,14 @@ class Meta:
     def validate_name(self, value):
         existing_slug = Experiment.objects.filter(slug=slugify(value))
         if existing_slug:
-            raise serializers.ValidationError("Duplicate Name Error")
+            raise serializers.ValidationError(
+                "This experiment name already exists."
+            )
 
-        return value
+        if slugify(value):
+            return value
+        else:
+            raise serializers.ValidationError("Invalid name selection.")
```

Comment:
The word 'selection' to me usually suggests you're picking from a list, rather than inputting text.  Maybe there's a way to phrase this that's less machine error-y and more humane?

Reply:
I changed it to.... "That's an invalid name."